[WTF] Use SAFE_FPRINTF / SAFE_PRINTF_TYPE for logging

AlexsanderDamaceno opened the pull request:
#### 69d9dbfea8961f5a8dd994fac877e36903b422e5
<pre>
[WTF] Use SAFE_FPRINTF / SAFE_PRINTF_TYPE for logging
<a href="https://bugs.webkit.org/show_bug.cgi?id=298450">https://bugs.webkit.org/show_bug.cgi?id=298450</a>

Reviewed by Adrian Perez de Castro.

Use SAFE_PRINTF_TYPE / SAFE_FPRINTF to have functions parameters applying the right conversions and type checks.

* Source/WTF/wtf/Assertions.h:

Canonical link: <a href="https://commits.webkit.org/300391@main">https://commits.webkit.org/300391@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/487a0582b30633604dbb84443cafa7929e38053f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122327 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42031 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32713 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128913 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/74426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42748 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50625 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93033 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/74426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125279 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34098 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109540 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73642 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33097 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27699 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72403 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/114475 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/103669 "Build was cancelled. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; run-api-tests (failure); re-run-api-tests (failure); Compiled WebKit (cancelled)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27907 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131654 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/120853 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49267 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37490 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101578 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49641 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105760 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101408 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46779 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24910 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/46105 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19348 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49124 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54867 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/151012 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48593 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/38630 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51943 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50274 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6034c6e2d67f50d9e469e3e6c4f609b33a3dd021. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50908

aperezdc commented on Source/WTF/wtf/Assertions.h:
```diff
@@ -708,12 +708,12 @@ static constexpr bool unreachableForValue = false;
 #define RELEASE_LOG_WITH_LEVEL_IF(isAllowed, channel, logLevel, ...) do { \
     if ((isAllowed) && LOG_CHANNEL(channel).level >= (logLevel)) \
-        SUPPRESS_UNCOUNTED_LOCAL os_log(LOG_CHANNEL(channel).osLogChannel, __VA_ARGS__); \
+        SUPPRESS_UNCOUNTED_LOCAL os_log(LOG_CHANNEL(channel).osLogChannel, SAFE_PRINTF_TYPE(__VA_ARGS__)); \
```
Same here, let's add a macro parameter for the format string to leave the format string out of `SAFE_PRINTF_TYPE()`.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ee65dbf598e445932ca9e774e5d17abe778bd55d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50908

aperezdc reviewed (approved):
@AlexsanderDamaceno This looks good to me. Could you remove the “Draft” status? Let's try to get someone from Apple to approve the changes, because we are also changing their logging implementation with this patch.

aperezdc commented:
There is one typo in the commit log message, please update it before landing: `paramenters` -> `parameters`.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0810dce3698ff3ce59aae70332c2b2f1791b62c0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50908

aperezdc commented on Source/WTF/wtf/Assertions.h:
```diff
@@ -725,7 +725,7 @@ static constexpr bool unreachableForValue = false;
     auto& logChannel = LOG_CHANNEL(channel); \
     if (logChannel.state != WTFLogChannelState::Off) \
-        __android_log_print(ANDROID_LOG_ ## priority, LOG_CHANNEL_WEBKIT_SUBSYSTEM, "[%s] " fmt, logChannel.name, ##__VA_ARGS__); \
+        __android_log_print(ANDROID_LOG_ ## priority, LOG_CHANNEL_WEBKIT_SUBSYSTEM, "[%s] " fmt, logChannel.name, __VA__OPT__(, SAFE_PRINTF_TYPE(__VA_ARGS__))); \
```
...and same here, remove the `,` before `__VA_OPT__`.

AlexsanderDamaceno committed:
[WTF] Use SAFE_FPRINTF / SAFE_PRINTF_TYPE for logging
https://bugs.webkit.org/show_bug.cgi?id=298450

Reviewed by Adrian Perez de Castro.

Use SAFE_PRINTF_TYPE / SAFE_FPRINTF to have functions parameters applying the right conversions and type checks.

* Source/WTF/wtf/Assertions.h:

Canonical link: https://commits.webkit.org/300391@main
